backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -384,6 +388,59 @@ public Response updateEventTypeBehaviors(@Context SecurityContext sec,
         return Response.ok().build();
     }
 
+    /**
+     * Appends the given behavior group to the specified event type. Instead of using a UUID for the event type, its
+     * name is used. This is due to the <a href="https://issues.redhat.com/browse/RHCLOUD-22367">RHCLOUD-22367</a>
+     * ticket, which was created by the frontend team, and requested to be able to use the name, because apparently
+     * the frontend is capable of sending that name right away. Instead, if we request a UUID for the event type, they
+     * need to fetch all the event types and filter them one by one.
+     * @param securityContext the security context to get the org id from.
+     * @param eventTypeName the name of the event type the behavior group is going to be attached to.
+     * @param behaviorGroupId the ID of the behavior group that, supposedly, they just created.
+     * @return a 204 response on success.
+     */
+    @PUT
+    @Path("/eventTypes/{eventTypeName}/behaviorGroups/{behaviorGroupId}")
+    @Consumes(APPLICATION_JSON)
+    @Produces(TEXT_PLAIN)
+    @Operation(summary = "Add a behavior group to the given event type.")
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
+    @APIResponse(responseCode = "204")
+    @Transactional
```

Comment:
Initiating a transaction here will put pressure on the DB while we may process requests that point to an unknown event type and won't trigger any write operation in the DB. I think this annotation can be removed.

Reply:
Removed! Thanks!